prusti-viper/src/encoder/mir/procedures/encoder/elaborate_drops/mod.rs

```diff
@@ -53,8 +53,11 @@ pub(super) fn elaborate_drops<'v, 'tcx: 'v>(
     }
 
     validate(&input_facts, &location_table, mir);
-    let drop_patch = self::mir_transform::run_pass(tcx, mir);
-    let mir = apply_patch(drop_patch, mir, &mut input_facts, &mut location_table);
+    // FIXME: this is super awkward; `run_pass` used to take an immutable ref,
+    // but now it takes a mutable ref, so the dirty fix is to clone.
+    let mut mir = mir.clone();
+    let drop_patch = self::mir_transform::run_pass(tcx, &mut mir);
+    let mir = apply_patch(drop_patch, &mir, &mut input_facts, &mut location_table);
```

Comment:
`apply_patch` internally clones the body. However, I am not sure whether it is possible to avoid one of the clones.

Reply:
I could avoid `apply_patch` by doing the patch at the end of `run_pass` (see the commented code out there). But then `&mut input_facts, &mut location_table` isn't passed in and so I was worried they would get out of sync

Reply:
`apply_patch` has to update the Polonius facts, so that is not an option. Let's keep the additional `clone` since it is definitely not the largest inefficiency in our code base.

Reply:
I think the clone is fine because (as far as I understand) it's still O(N) where N is the number of MIR bodies that we get from the compiler. It would be much worse if we were doing O(N * M) clones, with M the number of times we need to e.g. iterate over the body during encoding.